src/app/App.js

```diff
@@ -74,7 +74,7 @@ function App() {
       console.log("wishlist after remove", wishlist)
     } else {
       setWishlist([...wishlist, product]);
-      //why does the latest product not show up when added to wishlist?
+      //why does the latest product not show up when added to wishlist? 
       console.log("wishlist after adding", wishlist);
```

Comment:
Comment: Its because wishlist is referring to the value in `useState` prior to the re-render, so although you've called setWishlist - the application doesn't wait until the re-render occurs - it keeps on going!

If you wanted to see what the value of wishlist is after every time it changes, you could do...

```js
useEffect(() => {
  console.log('wishlist after render', wishlist)
},[wishlist])
```

That array as the second argument means 'if this value has updated after a re-render, I want you to run this useEffect!)' It also runs on page load (or more accurately 'component mount'). 

If you just want useEffect to run when the page loads, you can give it an empty array as the second argument!